### DISCUSSION TITLE
Looking to Get Best Experience on Surface Pro 8

### DISCUSSION THREAD

**Discussion by vistis** [linux-surface/linux-surface#1373] [Q&A]
Hi all! I have been using my Surface Pro 8 with GNOME on Arch for about 3 months now, and the Linux on Surface experience has been delightful, all thanks to the amazing work by `linux-surface`. But, I have found some aspects of the experience to be pretty underwhelming. So, I would really appreciate if you could help me. Also, please feel free to share your experiences with Linux on the Surface Pro 8, I would love to hear it!

Here are some of the things that bug me:
- **Battery life**: the machine can only last about 3-4 hours watching YouTube videos, which I find to be pretty underwhelming. I am not saying that it is bad, it can do a day of school, albeit barely, having only about 35% left after 5 hours of note taking and internet browsing. I have a hunch that it can do better than that though. This is with `auto-cpufreq` too, before installing it the battery was draining like crazy, only 2 hours of YouTube.
- **Thermal**: the machine tends to get pretty hot when doing anything slightly demanding. I feel like the fan rarely comes on, even when it gets to the point of being too hot to touch. And when it reaches a critical point, especially when playing a game, the machine just throttles like crazy with the fan cranked all the way up. I think it can be fixed by editing the fan curve, but I have not found a way to do it yet.
- **Touchscreen**: it is a miracle that it works in the first place, but, I do not know if it is an issue with GNOME or iptsd, sometimes touchscreen just would not register, notably hamburger menu buttons in GNOME apps just do not work when tapped on. It is also pretty hard to scroll, often accidentally triggering a tap or select while trying to scroll which is very annoying. And some other times, the app just decided to stop working with the touchscreen even though it was working fine a minute before, requiring a restart to restore the functionality.

Some important information that I feel like you should know:
- I am not very knowledgeable about Linux. I do know some basic stuff, but I would not define myself as an advance or even intermediate Linux user
- I am using GNOME on Wayland and I have a bunch of extensions installed (like `Dash to Dock`, `GSConnect`, `Media Control`, etc.)
- I use Celeste to backup my files, which I leave running in the background all the time alongside Mullvad VPN, Geary, Newsflash and Discord
- My machine has an Intel Core i7 with 16GB of RAM
- I have followed all the steps and recommendations provided by the linux-surface's Wiki (except setting up Secure Boot, which I just can't get working for some reason)

Thank you for reading! Also, if you have any suggestions that will improve the experience, please do share it. I along with the rest of the community will greatly appreciate it!

**Comment by StollD** [comment]
> * **Battery life**: the machine can only last about 3-4 hours watching YouTube videos, which I find to be pretty underwhelming. I am not saying that it is bad, it can do a day of school, albeit barely, having only about 35% left after 5 hours of note taking and internet browsing. I have a hunch that it can do better than that though. This is with `auto-cpufreq` too, before installing it the battery was draining like crazy, only 2 hours of YouTube.

Tbh, battery life issues always *need* comparisons with Windows. But in your case, have you enabled hardware accelerated video decoding?

> * **Thermal**: the machine tends to get pretty hot when doing anything slightly demanding. I feel like the fan rarely comes on, even when it gets to the point of being too hot to touch. And when it reaches a critical point, especially when playing a game, the machine just throttles like crazy with the fan cranked all the way up. I think it can be fixed by editing the fan curve, but I have not found a way to do it yet.

You should setup thermald, this will (try to) apply the same thermal profile that Windows uses.

> * **Touchscreen**: it is a miracle that it works in the first place, but, I do not know if it is an issue with GNOME or iptsd, sometimes touchscreen just would not register, notably hamburger menu buttons in GNOME apps just do not work when tapped on. It is also pretty hard to scroll, often accidentally triggering a tap or select while trying to scroll which is very annoying. And some other times, the app just decided to stop working with the touchscreen even though it was working fine a minute before, requiring a restart to restore the functionality.

If you are missing inputs you can try calibrating iptsd: https://github.com/linux-surface/iptsd/wiki/Calibrating-iptsd.

Freezing apps IIRC are a GNOME issue, and can sometimes be fixed by moving the mouse cursor or changing the active window with Alt+Tab.

But like you noticed, not all apps are really optimized for touch inputs and treat them like any other mouse cursor.

**Reply by vistis** [reply]
> Tbh, battery life issues always need comparisons with Windows. But in your case, have you enabled hardware accelerated video decoding?

Oh, I forgot to mention that I did not use Windows at all on the Surface Pro 8 beside updating the firmware with it, so I don't really have a comparison to work with. Do you have any tips to get better battery life on Linux?

And as for hardware accelerated video decoding, I do have it turned on in Firefox (my browser of choice), or at least have the `Use recommended performance settings` option enabled.

>You should setup thermald, this will (try to) apply the same thermal profile that Windows uses.

I already have `thermald` installed, I just forgot to mention that. It does fix the horrendous throttling issue, but I still feel like the fan does not come on much and it still get pretty hot at time. I have looked at the wiki to try to find an optimal profile for the Surface Pro 8, but did not have any luck. (I might try tinkering with the config later to find out)

>If you are missing inputs you can try calibrating iptsd: https://github.com/linux-surface/iptsd/wiki/Calibrating-iptsd.

Okay, I will try.
